plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +83,220 @@ protected DistributedQueryRunner createQueryRunner()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
     }
 
-    private String runAndGetId(String query)
+    @Test
+    public void testCreateTableAsSelect()
+            throws Exception
+    {
+        assertFileSystemAccesses("CREATE TABLE test_create_as_select AS SELECT 1 col_name",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(MANIFEST, OUTPUT_FILE_LOCATION), 1)
+                        .addCopies(new FileOperation(METADATA, OUTPUT_FILE_CREATE), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, OUTPUT_FILE_CREATE_OR_OVERWRITE), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, OUTPUT_FILE_CREATE_OR_OVERWRITE), 1)
+                        .addCopies(new FileOperation(METADATA, OUTPUT_FILE_LOCATION), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, OUTPUT_FILE_LOCATION), 1)
+                        .build());
+    }
+
+    @Test
+    public void testSelect()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_select AS SELECT 1 col_name", 1);
+        assertFileSystemAccesses("SELECT * FROM test_select",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
+    }
+
+    @Test
+    public void testSelectWithFilter()
+            throws Exception
     {
-        return getDistributedQueryRunner().executeWithQueryId(TEST_SESSION, query).getQueryId().getId();
+        assertUpdate("CREATE TABLE test_select_with_filter AS SELECT 1 col_name", 1);
+        assertFileSystemAccesses("SELECT * FROM test_select_with_filter WHERE col_name = 1",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
     }
 
-    private static OperationCounts getCounts()
+    @Test
+    public void testJoin()
```

Comment:
let's add a join to  a partitioned table as in https://github.com/trinodb/trino/issues/7502, https://github.com/trinodb/trino/issues/7999, since it seems it could exercise some code paths more.